File: drills/elixir/harnesses/nc76_kth_largest_stream.exs
stream = fn k, initial, added ->
  {_store, answers} =
    Enum.reduce(added, {Solution.new(k, initial), []}, fn value, {store, answers} ->
      store = Solution.add(store, value)
      {store, [Solution.kth(store) | answers]}
    end)

  Enum.reverse(answers)
end

[
  {"k = 3 over [4, 5, 8, 2] then 3, 5, 10, 9, 4", inspect([4, 5, 5, 8, 8]),
   inspect(stream.(3, [4, 5, 8, 2], [3, 5, 10, 9, 4]))},
  {"k = 1 over [] then 1, 2, 0", inspect([1, 2, 2]), inspect(stream.(1, [], [1, 2, 0]))},
  {"k = 2 over [] then 5, 5", inspect([nil, 5]), inspect(stream.(2, [], [5, 5]))},
  {"kth before anything is added", inspect(nil), inspect(Solution.kth(Solution.new(2, [])))}
]

File: drills/python/harnesses/nc76_kth_largest_stream.py
try:
    (KthLargest)
except NameError:
    raise Exception("__signature_mismatch__")

__results__ = []
def __case__(label, expected, actual):
    __results__.append([label, repr(expected), repr(actual)])

def __stream__(k, initial, added):
    store = KthLargest(k, initial)
    return [store.add(value) for value in added]

__case__("k = 3 over [4, 5, 8, 2] then 3, 5, 10, 9, 4", [4, 5, 5, 8, 8], __stream__(3, [4, 5, 8, 2], [3, 5, 10, 9, 4]))
__case__("k = 1 over [] then 1, 2, 0", [1, 2, 2], __stream__(1, [], [1, 2, 0]))
__case__("k = 2 over [] then 5, 5", [None, 5], __stream__(2, [], [5, 5]))
